packages/at_client/lib/src/transformer/response_transformer/notification_response_transformer.dart

```diff
@@ -62,15 +61,18 @@ class NotificationResponseTransformer
       var decryptedValue = await _getDecryptedValue(atKey, atKey.key);
       atNotification.key = '${atNotification.to}:$decryptedValue';
       return atNotification;
-    } else if ((atNotification.value.isNotNull) &&
-        (config.shouldDecrypt && atNotification.id != '-1') &&
+    }
+    if (atNotification.value.isNotNull &&
+        atNotification.id != '-1' &&
         // The shared_key (which is a reserved key) has different decryption process
         // and is not a user created key.
         // Hence do not decrypt if key's are reserved keys
         AtKey.getKeyType(atKey.toString()) != KeyType.reservedKey) {
-      // decrypt the value
-      atNotification.value =
-          await _getDecryptedValue(atKey, atNotification.value!);
+      // decrypt the notification value only if isEncrypted is not set to false
+      if (atNotification.isEncrypted != false) {
```

Comment:
If it's not false then we need to decrypt - but only if `config.shouldDecrypt` is true i.e. this line should read, I think, `if (atNotification.isEncrypted != false && config.shouldDecrypt)`